frontend/awx/resources/templates/hooks/useLaunchTemplate.tsx

```diff
@@ -0,0 +1,91 @@
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import { getJobOutputUrl } from '../../../views/jobs/jobUtils';
+import { requestGet } from '../../../../common/crud/Data';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { usePageAlertToaster, usePageNavigate } from '../../../../../framework';
+import type { JobTemplate } from '../../../interfaces/JobTemplate';
+import type { UnifiedJob } from '../../../interfaces/UnifiedJob';
+import type { WorkflowJobTemplate } from '../../../interfaces/WorkflowJobTemplate';
+import type { JobLaunch, WorkflowJobLaunch } from '../../../interfaces/generated-from-swagger/api';
+import { AwxRoute } from '../../../AwxRoutes';
+
+type Template = JobTemplate | WorkflowJobTemplate;
+type TemplateLaunch = JobLaunch & WorkflowJobLaunch;
+
+export function useLaunchTemplate() {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const postRequest = usePostRequest();
+  const alertToaster = usePageAlertToaster();
+  const pageNavigate = usePageNavigate();
+
+  return async (template: Template) => {
+    const launchEndpoint = getLaunchEndpoint(template);
+
+    if (!launchEndpoint) {
+      return Promise.reject(new Error('Unable to retrieve launch configuration'));
+    }
+
+    try {
+      const launchConfig = await requestGet<TemplateLaunch>(launchEndpoint);
+
+      if (canLaunchWithoutPrompt(launchConfig)) {
+        let launchJob;
+        if (template.type === 'job_template') {
+          launchJob = await postRequest(launchEndpoint, {});
+        } else if (template.type === 'workflow_job_template') {
+          launchJob = await postRequest(launchEndpoint, {});
+        }
+        navigate(getJobOutputUrl(launchJob as UnifiedJob));
+      } else {
+        if (template.type === 'job_template') {
+          pageNavigate(AwxRoute.TemplateLaunchWizard, { params: { id: template.id } });
+        } else if (template.type === 'workflow_job_template') {
+          pageNavigate(AwxRoute.WorkflowJobTemplateLaunchWizard, { params: { id: template.id } });
+        }
+      }
+    } catch (error) {
+      alertToaster.addAlert({
+        variant: 'danger',
+        title: t('Failed to launch template'),
+        children: error instanceof Error && error.message,
```

Comment:
You _could_ add a timeout property to this object so that the alert will disappear on its own.